### ISSUE TITLE
Fill Issue

### ISSUE THREAD

**Issue by dickinson0718** [dulimarta/spherical-easel#103]
If you
1. Create a line segment
2. Open the style panel, select the global panel and then choose a shade fill
3. Create a circle (a shaded circle is created)
4. Moving the circle causes the following error in the Javascript console:

```
 aria-label="pinia.js?v=449c678f:5508 Uncaught TypeError: Cannot read from private field
    at __typeError (chunk-TKZWA4SW.js?v=449c678f:4:9)
    at __accessCheck (chunk-TKZWA4SW.js?v=449c678f:11:62)
    at __privateGet (chunk-TKZWA4SW.js?v=449c678f:12:46)
    at Proxy.trigger (chunk-H2LCIYDK.js?v=449c678f:73:12)
    at Proxy.splice (chunk-H2LCIYDK.js?v=449c678f:103:10)
    at noTracking (chunk-GSZFOY4F.js?v=449c678f:1168:36)
    at Proxy.splice (chunk-GSZFOY4F.js?v=449c678f:1082:12)
    at Proxy.updateDisplay (Circle.ts:371:66)
    at Proxy.shallowUpdate (SECircle.ts:182:16)
    at Proxy.rotateSphere (se.ts:987:16) 
```

**Comment by dulimarta** [comment]
While reading the threads of https://github.com/vuejs/pinia/issues/1427, also found this https://github.com/vuejs/pinia/issues/1401 and its [workaround](https://codesandbox.io/s/pinia-type-script-private-fields-forked-8l8j82?file=/src/index.ts)

**Comment by dickinson0718** [comment]
I don't know how to recreate this particular way of getting the same error, but I was experimenting with circle concurrence and just deleted one of the three light-blue measured circles, then undone the delete, then as I was deleting another one (I think a non-current principle parent), this error occurred. (Edit:  This is repeatable as I tried the same exploration and got the same error, but if I keep trying to delete the circle I also a second slightly different error message
```
Uncaught TypeError: Cannot read from private field
    at __accessCheck (two__js.js?v=449c678f:33:11)
    at __privateGet (two__js.js?v=449c678f:36:3)
    at get ids (two__js.js?v=449c678f:4191:12)
    at Reflect.get (<anonymous>)
    at MutableReactiveHandler.get (chunk-GSZFOY4F.js?v=449c678f:1215:25)
    at Proxy.remove (two__js.js?v=449c678f:6262:37)
    at Proxy.remove (chunk-A5CSWPBR.js?v=449c678f:1842:17)
    at Proxy.removeFromLayers (Point.ts:216:21)
    at Proxy.removePoint (se.ts:624:23)
    at Proxy.wrappedAction (pinia.js?v=449c678f:5505:18)
```
and
```
pinia.js?v=449c678f:5508 Uncaught TypeError: Cannot read from private field
    at __accessCheck (two__js.js?v=449c678f:33:11)
    at __privateGet (two__js.js?v=449c678f:36:3)
    at get ids (two__js.js?v=449c678f:4191:12)
    at Reflect.get (<anonymous>)
    at MutableReactiveHandler.get (chunk-GSZFOY4F.js?v=449c678f:1215:25)
    at _Group.replaceParent (two__js.js?v=449c678f:6614:33)
    at _Group.RemoveChildren (two__js.js?v=449c678f:6045:21)
    at Children.dispatchEvent (two__js.js?v=449c678f:249:19)
    at Children.dispatchEvent (two__js.js?v=449c678f:1299:54)
    at Children.trigger (two__js.js?v=449c678f:255:31)
```

<img width="1342" alt="Image" src="https://github.com/user-attachments/assets/b1692c0d-fcef-483b-898a-c5b9479a0ed1" />

**Comment by dickinson0718** [comment]
Unfortunately, the fill issue and ellipse issue are fixed  (yay!) but the [second one](https://github.com/dulimarta/spherical-easel/issues/103#issuecomment-2959225165) -- the one with the concurrence of circles is not :-(

There is something else going on with concurrence of circle -- one of the intersections points between two measured circles is not registering.  I will investigate more tomorrow.

**Comment by dulimarta** [comment]
The previously attempted `patch-package` trick did not work because the patch was applied to `src/collection.js` but at runtime the actual function call to `get ids` refers to `build/two.module.js` (snippet from lines 4260-4289 are copied below):

```
// src/children.js
var _ids;
var Children = class extends Collection {
  constructor(children) {
    children = Array.isArray(children) ? children : Array.prototype.slice.call(arguments);
    super(children);
    __privateAdd(this, _ids, {});
    this.attach(children);
    this.on(Events.Types.insert, this.attach);
    this.on(Events.Types.remove, this.detach);
  }
  get ids() {
    return __privateGet(this, _ids);
  }
  attach(children) {
    for (let i = 0; i < children.length; i++) {
      const child = children[i];
      if (child && child.id) {
        this.ids[child.id] = child;
      }
    }
    return this;
  }
  detach(children) {
    for (let i = 0; i < children.length; i++) {
      delete this.ids[children[i].id];
    }
    return this;
  }
};

_ids = new WeakMap();
```

The two lines which do not make sense to me are the declaration of `_ids`, **outside the class scope**,
and he exception was thrown because of the following call

```
get ids() {
  return __privateGet(this, _ids)
}
```
which can be traced back to around line 50 of the same file:

```
var __accessCheck = (obj, member, msg) => {
  if (!member.has(obj))
    throw TypeError("Cannot " + msg);
};
var __privateGet = (obj, member, getter) => {
  __accessCheck(obj, member, "read from private field");
  return getter ? getter.call(obj) : member.get(obj);
};
```

The `_privateGet(this, _ids)` seems seems to imply that `_ids` was a property of `this` class but its actual declaration is outside the class scope. I tried to regenerate a new patch by moving `_ids` into the scope of `Collection` but this attempt did not fix the issue.

**Comment by dulimarta** [comment]
Commit 65fb243ed2464db7c263b0c0aa07807239f6ef74 is another attempt to fix this issue. Essentially, all `SExxx` objects are now added to reactive map after being marked with `markRaw`.

For instance, older snippet:
```
seLineMap.set(line.id, line);
seNodules.value.push(line as SENodule);
```
becomes
```
seLineMap.set(line.id, markRaw(line));
seNodules.value.push(markRaw(line));
```

**Comment by dickinson0718** [comment]
@dulimarta This is great! I was not able to duplicate any of the error that led to this issue. THANK YOU!!!
